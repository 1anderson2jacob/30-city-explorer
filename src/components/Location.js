import React, { Component } from 'react';
import superagent from 'superagent';
import Deck from './Deck';
import '../styles/location.scss';

export class Location extends Component {
  state = {
    userLocation: '',
    location: '',
    latitude: '',
    longitude: '',
    weather: [],
    yelp: [],
    meetup: [],
    movies: [],
    trails: [],
    isLoading: true
  }

  // ⭐️ React Functions ⭐️ 
  // ✅   goes off when page loads
  async componentDidMount() {
    this.getLocationDetails();
  };


  // ⭐️ Custom Functions ⭐️
  // ✅  gets location details and sets state
  getLocationDetails = async () => {
    const { location } = this.props.match.params;
    let data = await superagent.get(`https://city-explorer-backend.herokuapp.com/location`, { data: this.state.userLocation || location });
    const { formatted_query, latitude, longitude } = data.body;
    this.setState({
      isLoading: false,
      location: formatted_query,
      latitude,
      longitude,
      userLocation: ''
    });
    this.getResource('weather', data.body);
    this.getResource('movies', data.body);
    this.getResource('yelp', data.body);
    this.getResource('meetups', data.body);
    this.getResource('trails', data.body);
  }

  getResource = async (resource, location) => {
    let data = await superagent.get(`https://city-explorer-backend.herokuapp.com/${resource}`, { data: location })
    this.setState({ [resource]: data.body })
  }
  // ✅  handles new search input, updates state
  // ✅  changes state of location
  handleChange = e => {
    this.setState({ userLocation: e.target.value })
  }
  // ✅   updates URL
  handleSubmit = e => {
    e.preventDefault();
    this.props.history.push(`/${this.state.userLocation}`);
    this.getLocationDetails();

  }

  // ⭐️  Render functions below ⭐️
  // ✅  runs loading bar until gets state
  renderLocationDetails = () => {
    if (this.state.isLoading) return <h2>Loading...</h2>

    return (
      <div>
        <div className="map">
          <h2>{this.state.location}</h2>
          <img src={`https://maps.googleapis.com/maps/api/staticmap?center=${this.state.latitude}%2c%20${this.state.longitude}&zoom=12&size=400x300&maptype=roadmap&key=AIzaSyDp0Caae9rkHUHwERAFzs6WN4_MuphTimk`} alt="" />
        </div>
        <Deck details={this.state.weather} type={'weather'} />
        {/* <Deck details={this.state.movies} />
        <Deck details={this.state.yelp} />
        <Deck details={this.state.meetups} />
        <Deck details={this.state.trails} /> */}
      </div>
    )
  }
  // ⭐️  END ⭐️

  render() {
    return (
      <div className="location">
        <h4>City Explorer</h4>
        <form onSubmit={this.handleSubmit}>
          <div className="search">
            <input className='input' type="text" onChange={this.handleChange} placeholder="New location" value={this.state.userLocation} />
            <button>Explore!</button>
          </div>
        </form>
        {this.renderLocationDetails()}
      </div>
    );
  }
}

export default Location;
