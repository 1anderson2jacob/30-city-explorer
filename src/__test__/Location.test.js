import React from 'react';
import ReactDOM from 'react-dom';
import Enzyme, { shallow, render, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';

Enzyme.configure({ adapter: new Adapter() });
import Location from '../components/Location';

describe('running tests on city explore Location results', () => {
  it('is alive', () => {
    let component = shallow(<Location />);
    expect(component.find('h4').exists()).toBeTruthy();
  });
  it('renders correctly', () => {
    let render = renderer.create('<Location/>').toJSON();
    expect(render).toMatchSnapshot();
  });

});