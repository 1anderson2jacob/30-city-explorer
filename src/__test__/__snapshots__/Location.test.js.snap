// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`running tests on city explore Location results renders correctly 1`] = `"<Location/>"`;
